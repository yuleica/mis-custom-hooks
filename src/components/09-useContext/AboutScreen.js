import React, { useContext } from 'react';
import { UserContext } from './UserComponent';

export const AboutScreen = () => {

    const {user, setUser} = useContext(UserContext)

    const handleClick = () => {
        //objeto vac√≠o para limpiar la info
        setUser({});
    }

    return (
        <div>
            <h1>AboutScreen</h1>
            <hr />  

            <pre>
                {JSON.stringify(user, null, 3)}
            </pre>

            <button
                className="btn btn-warning"
                onClick={handleClick}
            >
                Logout
            </button>
        </div>
    )
}
