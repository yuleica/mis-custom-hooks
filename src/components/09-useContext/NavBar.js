import React from 'react';
import {Link, NavLink} from 'react-router-dom';

//link y navlink hace lo mismo...
//cuando uso navlink debo usar activeclassname
//cuando uso lin debo usar classname

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                <Link to="/" className="navbar-brand">useContext</Link>
                
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink exact activeClassName="active" to="/" className="nav-link active">Home</NavLink>
                        <NavLink exact activeClassName="active" to="/about" className="nav-link" >About</NavLink> 
                        <NavLink exact activeClassName="active" to="/login" className="nav-link" >Login</NavLink>
                    
                    </div>
                </div>
            </div>
        </nav>
    )
}
