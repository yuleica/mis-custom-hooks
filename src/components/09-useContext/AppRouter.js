import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';

//switch lo uso para indicar las rutas de lo componentes 
//a renderizar
export const AppRouter = () => {
    return (
        <Router>
            <div>

                <NavBar />

                <div className="container">

                    <Switch>

                        <Route exact path="/" component={HomeScreen} />

                        <Route exact path="/about" component={AboutScreen} />
                        <Route exact path="/login" component={LoginScreen} />
                        
                        <Redirect to="/" />
                    </Switch>

                </div>


            </div>
        </Router>

    )
}