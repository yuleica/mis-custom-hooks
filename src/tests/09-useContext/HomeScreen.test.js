import React from 'react';
import {mount} from 'enzyme';
import {HomeScreen} from '../../components/09-useContext/HomeScreen';
import { UserContext } from '../../components/09-useContext/UserComponent';

const user = {
    name: 'x',
    email: 'x@x.com'
}

const wrapper = mount(
    <UserContext.Provider value={{
        user

    }}>
        < HomeScreen />
    </UserContext.Provider>);
    

test('should show correctly', () => {
    
    expect(wrapper).toMatchSnapshot();

})

