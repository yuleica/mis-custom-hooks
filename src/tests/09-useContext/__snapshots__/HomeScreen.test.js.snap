// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "x",
   "email": "x@x.com"
}
  </pre>
</div>
`;
